solidity/contracts/modules/MultisigModule.sol

```diff
@@ -0,0 +1,349 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ External Imports ============
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+// ============ Internal Imports ============
+import {IMultisigModule} from "../../interfaces/IMultisigModule.sol";
+import {MessageV2} from "../libs/MessageV2.sol";
+import {MultisigModuleMetadata} from "../libs/MultisigModuleMetadata.sol";
+import {MerkleLib} from "../libs/Merkle.sol";
+
+/**
+ * @title MultisigModule
+ * @notice Manages an ownable set of validators that ECDSA sign checkpoints to
+ * reach a quorum.
+ */
+contract MultisigModule is IMultisigModule, Ownable {
+    // ============ Libraries ============
+
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using MessageV2 for bytes;
+    using MultisigModuleMetadata for bytes;
+    using MerkleLib for MerkleLib.Tree;
+
+    // ============ Mutable Storage ============
+
+    // The validator threshold for each remote domain.
+    mapping(uint32 => uint256) public threshold;
+
+    // The validator set for each remote domain.
+    mapping(uint32 => EnumerableSet.AddressSet) private validatorSet;
+
+    // A succinct commitment to the validator set and threshold for each remote
+    // domain.
+    mapping(uint32 => bytes32) public commitment;
```

Comment:
I don't think it's incompatible, just that you would need some routing module to redirect to the right contract based on origin domain.

My thinking was to push ISM iteration (aggregating ISMs, CCIP-read, etc) out of the V2 MVP. It should be pretty easy to iterate on this since the default ISM can be swapped out at any time. Ofc, some things will need to ossify before folks can start using sovereign consensus seamlessly

Reply:
happy to iterate on this in future

Reply:
It *could* but that would require the recipient contract implementing an interface as opposed to a storage variable

Reply:
public storage variable is arguably the same as implementing an interface but practically I see your point